.github/workflows/ci.yml

```diff
@@ -0,0 +1,392 @@
+          bundler-cache: true
+
+      - name: Run specs
+        run: bundle exec sass-spec.rb --dart .. $extra_args
+        working-directory: ./sass-spec
+        env:
+          extra_args: "${{ matrix.async_args }}"
+
+  dart_tests:
+    runs-on: "${{ matrix.os }}"
+
+    name: "Dart tests | Dart ${{ matrix.dart_channel }} | ${{ matrix.os }}"
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        dart_channel: [stable]
+        include:
+          - os: ubuntu-latest
+            dart_channel: dev
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Dart
+        uses: cedx/setup-dart@v2
+        with:
+          release-channel: "${{ matrix.dart_channel }}"
+
+      - name: Install app dependencies
+        run: dart pub get
+
+      - name: Build
+        run: dart pub run grinder pkg-standalone-dev
+
+      - name: Run tests
+        run: dart pub run test -p vm -x node
+
+  # Unit tests that use Node.js, defined in test/.
+  #
+  # The versions should be kept up-to-date with the latest LTS Node releases.
+  # They next need to be rotated April 2021. See
+  # https://github.com/nodejs/Release.
+  node_tests:
+    runs-on: "${{ matrix.os }}"
+
+    name: "Node tests | Dart ${{ matrix.dart_channel }} | Node ${{ matrix.node_version }} | ${{ matrix.os }}"
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        dart_channel: [stable]
+        # Run on Node 14 until laverdet/node-fibers#445 is fixed.
+        node_version: [14]
+        include:
+          # Include LTS versions on Ubuntu
+          - os: ubuntu-latest
+            dart_channel: stable
+            node_version: 12
+          - os: ubuntu-latest
+            dart_channel: stable
+            node_version: 10
+# TODO(nweiz): Re-enable these when dart-lang/sdk#44181 or dart-lang/test#1363 is fixed.
+#          - os: ubuntu-latest
+#            dart_channel: dev
+#            node_version: 14
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Dart
+        uses: cedx/setup-dart@v2
+        with:
+          release-channel: "${{ matrix.dart_channel }}"
+
+      - name: Install app dependencies
+        run: dart pub get
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '${{ matrix.node_version }}'
+
+      - name: Install node dependencies
+        run: npm install
+
+      - name: Build
+        run: dart pub run grinder before-test
+
+      - name: Run tests
+        run: dart pub run test -j 2 -t node
+
+  static_analysis:
+    name: Static analysis
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Dart
+        uses: cedx/setup-dart@v2
+
+      - name: Install app dependencies
+        run: dart pub get
```

Comment:
There's so much repetition in this file, it's really unfortunate that it doesn't support YAML references :confused:

Reply:
I totally agree with you on that point.